shaping/wrapping.go

```diff
@@ -362,6 +363,200 @@ func (r *runMapper) mapRun(runIdx int, run Output) {
 	}
 }
 
+// RunIterator defines a type that can incrementally provide shaped text.
+type RunIterator interface {
+	// Next returns the next logical run index, shaped run, and true if there are more runs,
+	// then it advances the iterator to the next element.
+	// Otherwise it returns an undefined index, an empty Output, and false.
+	Next() (index int, run Output, isValid bool)
```

Comment:
from the docs shouldn't `isValid` be called `hasNext` or similar?

Reply:
Totally correct, this is an oversight from a previous API design. Thanks for the catch!